packages/core/addon/navi-visualization-manifests/base.ts

```diff
@@ -0,0 +1,91 @@
+/**
+ * Copyright 2020, Yahoo Holdings Inc.
+ * Licensed under the terms of the MIT license. See accompanying LICENSE.md file for terms.
+ *
+ * Visualization Base Manifest File
+ * This file registers the visualization with navi
+ *
+ */
+import { assert } from '@ember/debug';
+import EmberObject from '@ember/object';
+import RequestFragment from 'navi-core/models/bard-request-v2/request';
+
+export default class NaviVisualizationBaseManifest extends EmberObject {
+  name!: string;
+  niceName!: string;
+  icon!: string;
+
+  /**
+   * checks if the request has single metric
+   */
+  hasSingleMetric(request: RequestFragment): boolean {
+    return request.metricColumns.length === 1;
+  }
+
+  /**
+   * checks if the request has no metrics
+   */
+  hasNoMetric(request: RequestFragment): boolean {
+    return request.metricColumns.length === 0;
+  }
+
+  /**
+   * checks if the request has some metrics
+   */
+  hasMetric(request: RequestFragment): boolean {
+    return !this.hasNoMetric(request);
+  }
+
+  /**
+   * checks if the request has single time bucket
+   */
+  hasInterval(request: RequestFragment): boolean {
+    const { timeGrain, interval } = request;
+
+    return !!(timeGrain && interval);
+  }
+
+  /**
+   * checks if the request has single time bucket
+   */
+  hasSingleTimeBucket(request: RequestFragment): boolean {
+    const { timeGrain, interval } = request;
```

Comment:
do these properties still exist on request v2?

Reply:
On the request fragment, yes for now
https://github.com/yahoo/navi/blob/431e2714462f222a9d9eff61a07d4404b9d848ae/packages/core/addon/models/bard-request-v2/request.ts#L194
https://github.com/yahoo/navi/blob/431e2714462f222a9d9eff61a07d4404b9d848ae/packages/core/addon/models/bard-request-v2/request.ts#L210